### PR TITLE
log,mon: use libfmt to format string and use range-based loop

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/log/Log.cc [hunk 1]
```diff
@@ -311,20 +313,16 @@ void Log::_flush(EntryVector& t, bool crash)
...
-    b += '\n';
-    int r = safe_write(m_fd, b.c_str(), b.size());
+    std::string b = fmt::format("{}\n", s);
+    int r = safe_write(m_fd, b.data(), b.size());
     if (r < 0)
       std::cerr << "problem writing to " << m_log_file << ": " << cpp_strerror(r) << std::endl;
   }
   if ((crash ? m_syslog_crash : m_syslog_log) >= 0) {
-    syslog(LOG_USER|LOG_INFO, "%s", s);
+    syslog(LOG_USER|LOG_INFO, "%s", s.data());
```
Comment (tchaikov) [inline src/log/Log.cc:325]: might need to ensure that `s` is nul terminated.

**Review by batrick — APPROVED** [review]

**Review by dillaman — APPROVED** [review]

**Comment by neha-ojha** [comment]
jenkins test api

**Review by lxbsz — APPROVED** [review]
LGTM.

**Review by neha-ojha — APPROVED** [review]

**Comment by harishmunjulur** [comment]
[wip-hmunjulu02-testing-2020-11-09-1705](https://trello.com/c/dcX5RS70/1065-wip-hmunjulu02-testing-2020-11-09-1705)

**Comment by tchaikov** [comment]
jenkins test api

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-11-14_09:04:49-rados-wip-kefu-testing-2020-11-14-1525-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-11-15_03:36:16-rados-wip-kefu-testing-2020-11-14-1526-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-11-15_12:08:30-rados-wip-kefu-testing-2020-11-14-1526-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-11-15_16:16:56-rados-wip-kefu-testing-2020-11-14-1526-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/48220
- https://tracker.ceph.com/issues/48030

**Pull request merged**